apps/src/code-studio/makerlab/setupPage.js

```diff
@@ -0,0 +1,207 @@
+import BoardController from '@cdo/apps/makerlab/BoardController';
+import React from 'react';
+import ReactDOM from 'react-dom';
+import classNames from 'classnames';
+
+const HIDDEN = 0;
+const WAITING = 1;
+const ATTEMPTING = 2;
+const SUCCEEDED = 3;
+const FAILED = 4;
+const CELEBRATING = 5;
+const STEP_STATUSES = [HIDDEN, WAITING, ATTEMPTING, SUCCEEDED, FAILED, CELEBRATING];
+
+const BoardSetupStatus = React.createClass({
+  getInitialState() {
+    return {
+      'status-is-chrome': WAITING,
+      'status-app-installed': WAITING,
+      'status-windows-drivers': WAITING,
+      'status-board-plug': WAITING,
+      'status-board-connect': WAITING,
+      'status-board-components': WAITING
+    }
+  },
+
+  render() {
+    return (
+        <div className="setup-status" style={{'fontSize': '26px'}}>
+          <SetupStep stepStatus={this.state['status-is-chrome']}
+                     stepId="is-chrome"
+                     stepName="Chrome version 33+"/>
+          <SetupStep stepStatus={this.state['status-app-installed']}
+                     stepId="app-installed"
+                     stepName="Chrome App installed"/>
+          <SetupStep stepStatus={this.state['status-windows-drivers']}
+                     stepId="windows-drivers"
+                     stepName="Windows drivers installed"/>
+          <SetupStep stepStatus={this.state['status-board-plug']}
+                     stepId="board-plug"
+                     stepName="Board plugged in"/>
+          <SetupStep stepStatus={this.state['status-board-connect']}
+                     stepId="board-connect"
+                     stepName="Board connectable"/>
+          <SetupStep stepStatus={this.state['status-board-components']}
+                     stepId="board-components"
+                     stepName="Board components usable"/>
+        </div>
+    );
+  },
+
+  hide(selector) {
+    this.setState({[`status-${selector}`]: HIDDEN});
+  },
+
+  fail(selector) {
+    this.setState({[`status-${selector}`]: FAILED});
+  },
+
+  spin(selector) {
+    this.setState({[`status-${selector}`]: ATTEMPTING});
+  },
+
+  succeed(selector) {
+    this.setState({[`status-${selector}`]: SUCCEEDED});
+  },
+
+  thumb(selector) {
+    this.setState({[`status-${selector}`]: CELEBRATING});
+  },
+
+  componentDidMount() {
+    const isChrome = !!window.chrome;
+    const gtChrome33 = isChrome && getChromeVersion() >= 33;
+
+    function getChromeVersion() {
+      const raw = navigator.userAgent.match(/Chrom(e|ium)\/([0-9]+)\./);
+      return raw ? parseInt(raw[2], 10) : false;
+    }
+
+    function isWindows() {
+      return navigator.platform.indexOf('Win') > -1;
+    }
+
+    if (isChrome) {
+      if (gtChrome33) {
+        this.succeed('is-chrome');
+      } else {
+        $('is-chrome').append("Your Chrome version is " + getChromeVersion() + ", please upgrade to at least version 33");
```

Comment:
Also, once this goes away your `stepId` property is entirely unused, and could be removed.